perf/change_point_detector.go

```diff
@@ -36,15 +52,37 @@ func NewMicroServiceChangeDetector(baseURL, token string) ChangeDetector {
 }
 
 func (spc *signalProcessingClient) DetectChanges(series []float64, ctx context.Context) ([]ChangePoint, error) {
-	changePoints := &struct {
-		ChangePoints []ChangePoint `json:"changePoints"`
+	jsonChangePoints := &struct {
+		ChangePoints []jsonChangePoint `json:"changePoints"`
 	}{}
 
-	if err := spc.doRequest(http.MethodPost, "change_points/detect", ctx, series, changePoints); err != nil {
+	if err := spc.doRequest(http.MethodPost, "change_points/detect", ctx, series, jsonChangePoints); err != nil {
 		return nil, errors.WithStack(err)
 	}
 
-	return changePoints.ChangePoints, nil
+	var result []ChangePoint
+	for _, point := range jsonChangePoints.ChangePoints {
+		mapped := ChangePoint{
+			Index: point.Index,
+			Info: AlgorithmInfo{
+				Name:    point.Algorithm.Name,
+				Version: point.Algorithm.Version,
+				Options: nil,
+			},
+		}
+
+		for k, v := range point.Algorithm.Configuration {
+			additionalOption := AlgorithmOption{
+				Name:  k,
+				Value: v,
+			}
+			mapped.Info.Options = append(mapped.Info.Options, additionalOption)
+		}
+
+		result = append(result, mapped)
```

Comment:
do you plan to change the results from the service to use the new format?

Reply:
Not at the moment; is there some compelling reason to do this everywhere vs what the service currently exposes?

Reply:
I think this is probably the best way to store the data, and given that you control the service on the other end, I think it probably makes sense to avoid reparsing all of your data, and these kinds of API changes are likely to be easier to make now rather than later.